rust/tools/authorization-logic/src/souffle/souffle_interface.rs

```diff
@@ -28,6 +29,7 @@ use crate::{
 /// Given a filename containing policy code this function parses it,
 /// constructs an AST, translates the AST to the datalogIR, and emits a string
 /// contianing souffle code.
+#[allow(dead_code)]
```

Comment:
StackOverflow makes it sound like there's a way to state that one wants all code/fns in a file to have this allow, but I tried at top of file and it didn't seem to work.

Reply:
What you have seems OK to me (listing it next to every function).

[This](https://stackoverflow.com/questions/25877285/how-to-disable-unused-code-warnings-in-rust) might be exactly the same stackoverflow you already looked at, but this one seems to suggest that you have to write `#![allow(dead_code)]` with the `!` for it to cover the whole crate. Since it says "crate-level" and not "file-level", maybe this means you have to put it in the `mod.rs` file rather than whichever file you are defining the function in. I did not try this myself, though!

Reply:
I think the 'top of file' one needs a 'bang'

```
#![allow(dead_code)]
#![allow(unused_variables)]
```

Still I believe these individual ones are better as the 'unused' message persists in code, and the warning is silenced.